### ISSUE TITLE
Duplicated map keys on snake case

### ISSUE THREAD

**Issue by danielsobrado** [Nooul/spring-boot-rest-api-helpers#20]
Hi,

The following function on [FilterService ](https://github.com/Nooul/spring-boot-rest-api-helpers/blob/master/src/main/java/com/nooul/apihelpers/springbootrest/services/FilterService.java)is creating duplicated maps when converting snake case to camel case, and both the snake case and camel case remain in the map, causing it to fail:

```java
    private HashMap<String, Object> convertToCamelCase(HashMap<String, Object> snakeCaseMap) {
        Set<String> keys = snakeCaseMap.keySet();
        HashMap<String, Object> camelCaseMap = new HashMap<>(snakeCaseMap);
        for (String key : keys) {
            Object val = snakeCaseMap.get(key);
            camelCaseMap.put(convertToCamelCase(key), val);
-->            camelCaseMap.remove(key);
        }
        return camelCaseMap;
    }
```

I included ```camelCaseMap.remove(key);``` to properly replace the key instead of adding a new one.

**Comment by zifnab87** [comment]
@danielsobrado good point - can you please make a pull request so I can merge it? Thanks

**Comment by danielsobrado** [comment]
Sure, let me create a test case and a PR.

**Comment by zifnab87** [comment]
this shouldn't be an issue any more: https://github.com/Nooul/spring-boot-rest-api-helpers/commit/13acefdd7a33da27bf3c00f845e673186f52c46e
